modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java

```diff
@@ -590,16 +590,23 @@ private void _upgradeNestedFields(
 		DDMFormField parentFieldSetDDMFormField) {
 
 		for (DDMFormField ddmFormField : ddmFormFields) {
+			DDMFormField fieldSetDDMFormField = _createFieldSetDDMFormField(
+				defaultLocale, ddmFormField.getName() + "FieldSet",
+				ListUtil.fromArray(ddmFormField), ddmFormField.isRepeatable());
+
 			if (ListUtil.isEmpty(ddmFormField.getNestedDDMFormFields())) {
-				parentFieldSetDDMFormField.addNestedDDMFormField(ddmFormField);
+				fieldSetDDMFormField.setProperty(
+					"rows", _getDDMFormFieldsRows(fieldSetDDMFormField));
+
+				ddmFormField.setNestedDDMFormFields(Collections.emptyList());
+				ddmFormField.setRepeatable(false);
+
+				parentFieldSetDDMFormField.addNestedDDMFormField(
+					fieldSetDDMFormField);
 
 				continue;
 			}
 
-			DDMFormField fieldSetDDMFormField = _createFieldSetDDMFormField(
-				defaultLocale, ddmFormField.getName() + "FieldSet",
-				ListUtil.fromArray(ddmFormField), ddmFormField.isRepeatable());
```

Comment:
If it works, you should probably fix the tests that are failing.

Reply:
Hi @carolmariaabb , with your code I get the following results with both examples: 
[Upgrade - Liferay DXP.webm](https://user-images.githubusercontent.com/29747802/214244758-acb4d403-e28e-4ee3-9829-f620cfe842f4.webm)

I'm not sure if the one with parent - child - grandchild elements is ok. What do you think?

Reply:
@crodriguezy Maybe it's better to open a PTR to discuss the most appropriate behavior.

In addition, since the behavior has changed, two tests are now failing: `DDMDataDefinitionConverterTest` and `JournalConverterUtilTest`. It would be nice to ask the echo team to review this change, as one of them is related to the `Journal` module.

Reply:
Hi @carolmariaabb , ok, I will open the PTR ASAP. As for the echo team, I already sent them the PR but they told me it was not their component, they just provided some help when needed back in those days. 
Thanks

Reply:
PTR created: https://issues.liferay.com/browse/PTR-3646
Could you please move it quickly so we can find out a solution for the customer sooner than later? Thanks